capstone/scripts/convert_s3.py

```diff
@@ -356,66 +354,29 @@ def copy_volume_pdf(
 # Case-specific helper functions
 
 
-def case_content_exists_in_s3(
-    bucket: str, key: str, file: tempfile.TemporaryFile
-) -> bool:
-    """
-    Check whether file with the same content already exists in S3 using
-    ChecksumSHA256 hash comparison.
-    """
+def fetch_s3_files(bucket: str, key: str) -> dict:
     try:
-        # Calculate the SHA256 hash of the file content
-        file.seek(0)
-        file_data = file.read()
-        hash_object = hashlib.sha256(file_data)
-        sha256_hash = base64.b64encode(hash_object.digest()).decode()
-
-        # Get the object's metadata
-        response = s3_client.get_object_attributes(
-            Bucket=bucket, Key=key, ObjectAttributes=["Checksum"]
-        )
-
-        existing_hash = response.get("Checksum", {}).get("ChecksumSHA256")
-
-        # Compare the hashes
-        return existing_hash == sha256_hash
-
+        s3_contents_hash = {}
+        response = s3_client.list_objects_v2(Bucket=bucket, Prefix=key)
     except ClientError as err:
-        if err.response["Error"]["Code"] == "404":
-            return False
-        else:
-            raise Exception(f"Cannot check file {bucket}/{key}: %s" % err)
-
-
-def build_unique_case_name(
-    first_page: str, bucket: str, case_prefix: str, file: tempfile.TemporaryFile
-) -> str:
-    """
-    Create unique case name based on first page
-    """
-    suffix = 1
-    first_page = int(first_page)
-    case_name = f"{first_page:04d}-{suffix:02d}"
-
-    while case_name_exists(case_name, bucket, case_prefix):
-        if case_content_exists_in_s3(bucket, f"{case_prefix}/{case_name}.json", file):
-            print(f"Skipping {case_prefix}/{case_name}. Duplicate content.")
-            return None
-        suffix += 1
-        case_name = f"{first_page:04d}-{suffix:02d}"
-
-    return case_name
+        raise Exception(f"Cannot list objects {bucket}/{key}: %s" % err)
+    if "Contents" not in response:
+        return s3_contents_hash
+    else:
+        for case in response["Contents"]:
+            case_file_name = case["Key"].split("/")[-1]
+            # Get the object's metadata
+            try:
+                response = s3_client.get_object_attributes(
```

Comment:
The checksum should be in the original list response https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html , so we shouldn't need an additional API call per file.

Reply:
I believe that the checksum algorithm is in the response, but the checksum is not. There is some chatter about the ETag being used for this, but according to the documentation:
“The ETag may or may not be an MD5 digest of the object data. Whether or not it is depends on how the object was created and how it is encrypted as described below”
Part of that is determined by size of the object, among other criteria (although I think it’s extremely unlikely we’ll hit the size limitation), so seems risky to use. That’s why I chose to do a second query. Does that seem misguided?
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html

Reply:
Ah got it, that's annoying -- thanks for checking. I think if speed of recreating a volume became an issue, switching to ETags might be a good idea. But you've also convinced me I'm overoptimizing, so what you have makes sense to me.